### PR TITLE
add haskell support

### PR DESCRIPTION
**Pull request by quangIO**

### CODE REVIEW

File: tree-sitter-langs.el [hunk 1]
```diff
@@ -92,6 +92,7 @@ See `tree-sitter-langs-repos'."
                 (css-mode        . css)
                 (elm-mode        . elm)
                 (go-mode         . go)
+                (haskell-mode         . haskell)
```
Comment (jcs090218) [inline tree-sitter-langs.el:95]: You might want to align this? :)

**Comment by ubolonton** [comment]
Haskell grammars seem to be broken on some platforms.
I'll check this after the CI pipelines are fixed. They are currently broken after this repo is moved to an org.

**Comment by quangIO** [comment]
On MacOS, I think the CI also need to use gcc instead of clang. E.g. `brew install gcc && ln -sf /usr/local/bin/gcc-11 /usr/local/bin/gcc`

**Comment by iyefrat** [comment]
This PR seems to have passed CI. Has the problem been fixed or is this a false positive or something?

Also @quangIO there's a merge conflict now.

Thanks for submitting this! Can't wait for it to get merged.

**Comment by ubolonton** [comment]
Yeah, it passed when CI was not properly set up to fail on errors.

**Comment by ZachFontenot** [comment]
Hey there, I haven't looked into the particulars, but maybe this PR https://github.com/tree-sitter/tree-sitter-haskell/pull/62 has made it easier to build this?

**Comment by ubolonton** [comment]
Thanks for the info. I added it in #70.

**Comment by unhammer** [comment]
It's missing the queries and entry in `tree-sitter-langs--init-major-mode-alist` though. I just tried it, and it seems to work, even handles TemplateHaskell better than the regular emacs haskell-mode.

**Pull request closed without merging**